internal/states/statefile/read_test.go

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MPL-2.0
+
+package statefile
+
+import (
+	"errors"
+	"os"
+	"testing"
+)
+
+func TestReadErrNoState(t *testing.T) {
+	emptyFile, err := os.Open("testdata/read/empty")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer emptyFile.Close()
+
+	_, err = Read(emptyFile)
+	if !errors.Is(err, ErrNoState) {
+		t.Fatalf("expected ErrNoState, got %T", err)
+	}
```

Comment:
can you split those into two different cases?

Reply:
Done.